sshd_config

```diff
@@ -104,6 +105,11 @@ Ciphers aes256-gcm@openssh.com,aes128-gcm@openssh.com
 HostKeyAlgorithms rsa-sha2-512,rsa-sha2-256,ssh-ed25519
 MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com
 
+# only use host keys with secure HostKeyAlgorithms
+# omit the ECDSA key since it is only used in conjunction with the NIST P-curves
+HostKey /etc/ssh/ssh_host_ed25519_key
+HostKey /etc/ssh/ssh_host_rsa_key
```

Comment:
I believe it is not necessary to include the rsa_key. Here is the reason:
1. The strength of the rsa_key may vary
2. Providing only the ed25519_key should not harm the compatibilities of this project. Thus, I prefer to reduce the attack surface
3. Time to create an SSH connection may also benefit by avoiding the rsa_key since it requires more bits to be sent

P.S. Point 3 is really not a good point, but the tune of this project is quite like that ;-)

Reply:
Just in case you have missed that, here is the suggestion from the Redhat Inc.
Refer to: [access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/configuring_basic_system_settings/assembly_using-secure-communications-between-two-systems-with-openssh_configuring-basic-system-settings#making-openssh-more-secure_assembly_using-secure-communications-between-two-systems-with-openssh](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/configuring_basic_system_settings/assembly_using-secure-communications-between-two-systems-with-openssh_configuring-basic-system-settings#making-openssh-more-secure_assembly_using-secure-communications-between-two-systems-with-openssh)

Reply:
Apologies for the delay. I have just been way too busy recently.

Yeah I made the change after seeing this page. My impression reading the text is that RH is giving an example of how to disable certain host keys, not recommending it to be done like that.

The other thing is, while RSA key can be weaker than ed25519, it can also be equivalently strong (3072 bits) or stronger (4096 bits). It is also required in some cases for FIPS compliance. I am therefore hesitant to outright exclude it.